homeassistant/components/netgear_lte/config_flow.py

```diff
@@ -0,0 +1,94 @@
+
+from typing import Any
+
+from aiohttp.cookiejar import CookieJar
+from eternalegypt import Error, Modem
+from eternalegypt.eternalegypt import Information
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import CONF_HOST, CONF_PASSWORD
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+
+from .const import DEFAULT_HOST, DOMAIN, LOGGER, MANUFACTURER
+
+
+class NetgearLTEFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Netgear LTE."""
+
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
+        """Import a configuration from config.yaml."""
+        return await self.async_step_user(
+            user_input={
+                CONF_HOST: config[CONF_HOST],
+                CONF_PASSWORD: config[CONF_PASSWORD],
+            }
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input:
+            host = user_input[CONF_HOST]
+            password = user_input[CONF_PASSWORD]
+
+            try:
+                info = await self._async_validate_input(host, password)
+            except InputValidationError as ex:
+                errors["base"] = ex.base
+            else:
+                await self.async_set_unique_id(info.serial_number)
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
+                    title=f"{MANUFACTURER} {info.items['general.devicename']}",
+                    data={CONF_HOST: host, CONF_PASSWORD: password},
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self.add_suggested_values_to_schema(
+                vol.Schema(
+                    {
+                        vol.Required(CONF_HOST): str,
+                        vol.Required(CONF_PASSWORD): str,
+                    }
+                ),
+                user_input or {CONF_HOST: DEFAULT_HOST},
+            ),
+            errors=errors,
+        )
+
+    async def _async_validate_input(self, host: str, password: str) -> Information:
+        """Validate login credentials."""
+        websession = async_create_clientsession(
+            self.hass, cookie_jar=CookieJar(unsafe=True)
+        )
+
+        modem = Modem(
+            hostname=host,
+            password=password,
+            websession=websession,
+        )
+        try:
+            await modem.login()
+        except Error as ex:
+            raise InputValidationError("cannot_connect") from ex
+        except Exception as ex:
+            LOGGER.exception("Unexpected exception")
+            raise InputValidationError("unknown") from ex
+        return await modem.information()
```

Comment:
Can this raise exceptions?

Reply:
login swallows asyncio.TimeoutError, ClientError, Error making it very broad. Only a key error might happen but then I think we would want the stack trace to diagnose. Calling out of the block could of course happen but unlikely?

Reply:
Only a key error might happen but then I think we would want the stack trace to diagnose. We would not get the trace with the key error since that is reraised in the package as the same error as failure to connect. Unless you want to log except for cannot connect as well?

Reply:
I'm referring to the `modem.information` call. Is that what you're talking about?

Reply:
Correct

Reply:
All possible errors should be caught in the config flow just as we do for the login call.